src/clib/pio_darray_int.c

```diff
@@ -1434,12 +1466,15 @@ int pio_read_darray_nc_serial(file_desc_t *file, io_desc_t *iodesc, int vid,
                 }
                 LOG((3, "maxregions = %d tmp_bufsize = %d", maxregions, tmp_bufsize));
 
+                /* Now get each region of data. */
                 loffset = 0;
                 for (int regioncnt = 0; regioncnt < maxregions; regioncnt++)
                 {
+                    /* Get pointer where data should go. */
                     bufptr = (void *)((char *)iobuf + tsize * loffset);
                     regionsize = 1;
 
+                    /* ??? */
```

Comment:
The decomposition may not use all of the active io tasks rtask here is the io task rank and ios->num_iotasks is the number of iotasks actually used in this decomposition.